Lib/mastermind.rb

```diff
@@ -1,5 +1,25 @@
+module Escape
+  def leave
+      puts "Are you sure you want to quit? Type \"y\" to quit or \"r\" to restart."
+        if gets.chomp.downcase == "r"
+          load 'mastermind.rb' 
+        else
+          abort("Thanks for playing!")
+        end
+    end
+
+  def restart
+    puts "Would you like to quit or restart? Type \"q\" to quit or \"r\" to restart."
+        if gets.chomp.downcase == "r"
+          load 'mastermind.rb' 
+        else
+          abort("Thanks for playing!")
+        end
+      end
+end
```

Comment:
I would rename this module to something like "Commands"

Reply:
On the lines where you `load 'mastermind.rb` I would extract those to a constant so if your path ever changes for `mastermind.rb` you only have to change it in one place.